x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -468,6 +485,33 @@ export class Embeddable
         this.input.onTableRowClick(event.data as unknown as LensTableRowContextMenuEvent['data']);
       }
     }
+
+    // TODO - inject this from datatable_visualization/visualization
+    const onEditAction = (state, event) => {
+      switch (event.data.action) {
+        case 'sort':
+          return {
+            ...state,
+            sorting: {
+              columnId: event.data.columnId,
+              direction: event.data.direction,
+            },
+          };
+        };
+    };
+
+    if (isLensEditEvent(event)) {
+      if (!this.savedVis) return;
+
+      const newSavedVis = JSON.parse(JSON.stringify(this.savedVis));
+      newSavedVis.state.visualization = onEditAction(this.savedVis.state.visualization, event);
```

Comment:
Not sure if there's a better way to do this....

Reply:
I don't see a reason why we couldn't mutate the existing this.savedVis - the embeddable is created and destroyed with the dashboard so it won't leak out of it

Reply:
Works for me!

Reply:
Update: `savedVis.state` is an object with immutable props, so we have to dance a bit.

<img width="514" alt="Screen Shot 2021-11-09 at 3 39 12 PM" src="https://user-images.githubusercontent.com/315764/141145398-4c1f77c7-cff7-44dd-86ff-40924e6baeb4.png">